src/server.cpp

```diff
@@ -222,12 +222,9 @@ Server::Server(
 	infostream<<"- world:  "<<m_path_world<<std::endl;
 	infostream<<"- game:   "<<m_gamespec.path<<std::endl;
 
-	// Initialize default settings and override defaults with those provided
-	// by the game
-	set_default_settings(g_settings);
-	Settings gamedefaults;
-	getGameMinetestConfig(gamespec.path, gamedefaults);
-	override_default_settings(g_settings, &gamedefaults);
+	// Create world if it doesn't exist
+	if(!initializeWorld(m_path_world, m_gamespec.id))
+		throw ServerError("Failed to initialize world");
```

Comment:
I agree with this (no need for an exception... just msg and abort)

In general I think that exceptions are used in a way that is not "good practice" throughout minetest. We shouldn't be adding/throwing more exceptions. All exceptions should follow "termination semantics" (rather than "resumption semantics" as they are in many instances throughout minetest) and in that case you may as well just... well, print a msg and abort